server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -283,12 +283,23 @@ public static class Builder extends FieldMapper.Builder {
 
         final TextParams.Analyzers analyzers;
 
-        public Builder(String name, IndexAnalyzers indexAnalyzers) {
-            this(name, IndexVersion.current(), indexAnalyzers);
+        public Builder(String name, IndexAnalyzers indexAnalyzers, boolean isSyntheticSourceAlwaysUsed) {
```

Comment:
Maybe rename to `synthetcSourceEnabledViaIndexMode`? To indicate that synthetic source is enabled because the index mode enables it.

Reply:
I was trying to express that and your suggestion is so much more clear!